zproject/backends.py

```diff
@@ -1211,6 +1214,9 @@ def social_associate_user_helper(backend: BaseAuth, return_data: Dict[str, Any],
         # we construct the full name from them.
         return_data["full_name"] = f"{first_name} {last_name}".strip()  # strip removes the unnecessary ' '
 
+    if user_avatar_url:
+        return_data["user_avatar_url"] = user_avatar_url
```

Comment:
`user_avatar_url` is declared as `str` type in ExteralAuthDataDict, but here the value can be None I think because of `user_avatar_url = kwargs['response'].get('avatar_url')`.